ceph-releases/luminous/centos/7/daemon/Dockerfile

```diff
@@ -8,30 +8,33 @@ MAINTAINER Sébastien Han "seb@redhat.com"
 ENV CEPH_VERSION luminous
 ENV CONFD_VERSION 0.10.0
 
-# Install Ceph
-RUN rpm --import 'https://download.ceph.com/keys/release.asc'
-RUN rpm -Uvh http://download.ceph.com/rpm-${CEPH_VERSION}/el7/noarch/ceph-release-1-1.el7.noarch.rpm
-RUN yum install -y epel-release
-ADD ganesha.repo /etc/yum.repos.d/
-RUN yum clean all
-ARG PACKAGES="unzip ceph-mon ceph-osd ceph-mds ceph-mgr ceph-base ceph-common ceph-radosgw rbd-mirror device-mapper sharutils etcd kubernetes-client e2fsprogs s3cmd wget nfs-ganesha nfs-ganesha-ceph nfs-ganesha-rgw"
-ARG PURGES="/tmp/* /var/tmp/* /usr/lib/{dracut,locale,systemd,udev} /usr/bin/hyperkube /usr/bin/etcd /usr/bin/systemd-analyze /etc/{udev,selinux} /usr/lib/{udev,systemd}"
-RUN yum install -y $PACKAGES && rpm -q $PACKAGES && yum clean all && rm -rf $PURGES
-
-# Install confd
-ADD https://github.com/kelseyhightower/confd/releases/download/v${CONFD_VERSION}/confd-${CONFD_VERSION}-linux-amd64 /usr/local/bin/confd
-RUN chmod +x /usr/local/bin/confd && mkdir -p /etc/confd/conf.d && mkdir -p /etc/confd/templates
-
-# Download forego
-RUN wget -O /forego.tgz 'https://bin.equinox.io/c/ekMN3bCZFUn/forego-stable-linux-amd64.tgz' && \
-  cd /usr/local/bin && tar xfz /forego.tgz && chmod +x /usr/local/bin/forego && rm /forego.tgz
-
-# Add s3cfg file
 ADD s3cfg /root/.s3cfg
 
 # Add bootstrap script, ceph defaults key/values for KV store
 ADD *.sh ceph.defaults check_zombie_mons.py ./osd_scenarios/* entrypoint.sh.in disabled_scenario /
 
+ARG PACKAGES="unzip ceph-mon ceph-osd ceph-mds ceph-mgr ceph-base ceph-common ceph-radosgw rbd-mirror device-mapper sharutils etcd kubernetes-client e2fsprogs s3cmd wget nfs-ganesha nfs-ganesha-ceph nfs-ganesha-rgw"
+
+# Install Ceph
+ADD ganesha.repo /etc/yum.repos.d/
+
+RUN rpm --import 'https://download.ceph.com/keys/release.asc' && \
+  rpm -Uvh http://download.ceph.com/rpm-${CEPH_VERSION}/el7/noarch/ceph-release-1-1.el7.noarch.rpm && \
+  yum install -y epel-release && \
+  yum clean all && \
+  yum install -y $PACKAGES && \
+  rpm -q $PACKAGES && \
+  yum clean all && \
+  \
+  # Install confd
+  wget -O /usr/local/bin/confd "https://github.com/kelseyhightower/confd/releases/download/v${CONFD_VERSION}/confd-${CONFD_VERSION}-linux-amd64" && \
+  chmod +x /usr/local/bin/confd && mkdir -p /etc/confd/conf.d && mkdir -p /etc/confd/templates && \
+  \
+  # Download forego
+  wget -O /forego.tgz 'https://bin.equinox.io/c/ekMN3bCZFUn/forego-stable-linux-amd64.tgz' && \
+  cd /usr/local/bin && tar xfz /forego.tgz && chmod +x /usr/local/bin/forego && rm /forego.tgz && \
+  bash /clean_container.sh && rm /clean_container.sh
+
 # Modify the entrypoint
 RUN bash "/generate_entrypoint.sh" && \
```

Comment:
can't we include this  `RUN` in the previous `RUN`?

Reply:
No, there is no particular gain in space and for the semantic, I do prefer keeping the install/clean in a single run and the add of the entrypoint in a separate one.